ably-ios/ARTAuth.m

```diff
@@ -274,6 +275,11 @@ - (void)setProtocolClientId:(NSString *)clientId {
 
 - (void)setTokenDetails:(ARTAuthTokenDetails *)tokenDetails {
     _tokenDetails = tokenDetails;
+    // If a token has been setted and the current Auth method is Basic then for now on use Token Authentication
```

Comment:
I don't understand this comment.  If a valid token is set, why not always use token auth from that point forwards?  Why is the current token method even relevant?